airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -136,27 +148,75 @@ def _create_prepared_request(
 
         return prepared_request
 
-    def _send_with_retry(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+    @cached_property
+    def _max_retries(self) -> int:
+        max_retries = None
+        if self._disable_retries:
+            max_retries = 0
+        else:
+            max_retries = self._error_handler.max_retries
+        return max_retries if max_retries is not None else self._DEFAULT_MAX_RETRY
+
+    @cached_property
+    def _max_time(self) -> int:
+        return self._error_handler.max_time if self._error_handler.max_time is not None else self._DEFAULT_MAX_TIME
+
+    @cached_property
+    def _factor(self) -> float:
+        factor: Optional[float] = None
+        for backoff_strategy in self._backoff_strategies:
+            if hasattr(backoff_strategy, "retry_factor") and backoff_strategy.retry_factor is not None:
```

Comment:
It seems like we are still relying on the declarative interfaces here. Should we have this `factor` as part of the `airbyte_cdk.sources.streams.http` package?

Reply:
Follow-up comment: The hasattr exists here because retry_factor is not enforced by the interface and is only used by exponential backoffs.

Reply:
What it means to be is that we got our abstractions wrong and we are at risk of a breaking change once we have to update this concept. 

The easy solution I see is to have `BackoffHandler` implement `def handle(?) -> Callable[[request, request_kwargs, log_formatter], Response]` or something like that and stop exposing  `max_tries`, `max_time` and `factor`. This though is a bigger change today. I'm fine with pushing this breaking change for later assuming the risk of occurrence is low and the blast radius is somewhat contained

Reply:
I agree with Maxime that the abstraction looks off. Why can't we rely on the backoff strategie's `backoff_time` to determine how long to wait before retrying? 

The declarative framework already supports exponential backoff without needing to leak the factor abstraction to the requester so we should be able to do the same here

Reply:
womp. looking at the http requester again, I see that the factor is indeed leaking because of the default value https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py#L504C9-L504C24

Reply:
I think it's more than default value here: it is the fact that we can have multiple backoff strategies that relies on different parameters that we expose. Hence, the implementation of the backoff strategy are not well encapsulated. We could encapsulate that in the HttpClient if we wanted

Reply:
@tolik0 @pnilan is there a reason we couldn't just use the [user_backoff_handler](https://github.com/airbytehq/airbyte/pull/38906/files#diff-6082346b7f2b198151e1a7b4629dd8428ab9330d225667b3dc5259e39a48434eR208)?

I wouldn't expect the default to be required since a backoff strategy should always be defined